enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,181 @@
+Key names must be in the format `ack-X.Y-<description>` where `X.Y` is the major/minor version number and `<description>` is words describing to what the administrator is agreeing.
+For example, a key name beginning with `ack-4.8` would only apply to an OCP cluster currently at version 4.8.
+If running within the applicable version, CVO will check a newly defined configmap for the key.
+The key represents a specific upgrade blocking condition that must be acknowledged before a minor level upgrade is allowed.
+An administrator acknowledges a condition by adding the corresponding key with a value of `true` to the configmap.
+Values other than `true` will be ignored and therefore not clear the upgrade block.
+The keys will be named to make it explicitly clear what the administrator is acknowledging and thereby agreeing to.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to the next minor release until it finds each of the applicable predefined keys in the configmap with a value of `true`.
+For each key blocking an upgrade, CVO will generate a detailed message containing the key and pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion, i.e. entering the corresponding key into the configmap with a value of `true`.
+
+The new configmap `upgrade-ack` will be installed as part of the release in the namespace `openshift-cluster-version`.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+  name: upgrade-ack
+  namespace: openshift-cluster-version
+```
+It will initially have no keys since the administrator is expected to add the relevant key as part of acknowledgement.
+
+### 4.8.z Specific Implementation
+
+Clusters upgraded to 4.9 will no longer have access to these [APIs][k8s-1.22-removed-apis].
+This new upgrade blocking mechanism will be used in an attempt to require administrators to read detailed documentation regarding the removed APIs and the possible impact.
+This OCP documentation will list which APIs have been removed and how an administrator checks their cluster for use of those APIs.
+
+4.8.z versions that are upgradeable to 4.9 will contain this enhancement and have the configmap `openshift-cluster-version/upgrade-ack`.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to 4.9 until the administrator has added the key/value pair `ack-4.8-apis-being-removed-in-4.9: "true"` to the configmap.
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  annotations:
+    ack-4.8-apis-being-removed-in-4.9: "true"
+  name: upgrade-ack
+  namespace: openshift-cluster-version
+```
+Until this is done the `Conditions` console page will display an `Upgradeable` condition message directing the administrator to the relevant OCP documentation.
+
+If the administrator wants to upgrade to 4.9 without performing the acknowledgment `force` can be used.
```

Comment:
Cloud credential operator has done something similar to this where if you're using manual mode, where an administrator is responsible for managing all credentails/permissions, we require the admin to annotate every CredentailsRequest to indicate they've done the work to prepare for an upgrade.

@akhil-rane @joelddiaz would we be interested in using this approach instead? It implies an "ack" key be automatically present for every minor version for clusters in manual mode.

Reply:
I think CCO current implementation is nice because the "ack" is in context of the CredentialsRequest, while they could certainly patch them without looking at the context of the actual request it sort of puts it in the admin's face.

Reply:
IIRC, you only annotate the CCO's config CR, not the individual CredentialsRequest CRs